core/util.go

```diff
@@ -478,6 +478,36 @@ func VerifyCSR(csr *x509.CertificateRequest) error {
 	return errors.New("Unsupported CSR signing algorithm")
 }
 
+var (
+	oidExtensionRequest = asn1.ObjectIdentifier{1, 2, 840, 113549, 1, 9, 14}
+	oidSubjectAltName   = asn1.ObjectIdentifier{2, 5, 29, 17}
+)
+
+// ExtensionsFromCSR extracts the set of requested extensions from a CSR
+func ExtensionsFromCSR(csr *x509.CertificateRequest) map[string][]byte {
+	extensionMap := map[string][]byte{}
+	for _, attr := range csr.Attributes {
```

Comment:
Nope, those are CSR extensions, not certificate extensions.  The way you request extension in a CSR is with the extensionRequest attribute.  

> attributes is a collection of attributes providing additional
> information about the subject of the certificate.  ...  Another example is information to
> appear in X.509 certificate extensions (e.g. the
> extensionRequest attribute from PKCS #9).

You can see `crypto/x509` builds the `DomainNames` list [in the same way](https://golang.org/src/crypto/x509/x509.go#L1988).